contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +252,86 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
+	// commitScriptTimeLockTaproot is a function that returns the timelock
+	// of a non-htlc commitment output.
+	//
+	// TODO(proofofkeags): move this
+	// NOTE FOR REVIEWERS: This function does not really belong here. That
+	// said, I don't know where it should go. If you have opinions, please
+	// comment.
+	commitScriptTimeLockTaproot := func(script []byte) fn.Option[int32] {
+		tok := txscript.MakeScriptTokenizer(0, script)
+		tok.Next() // PUSH <key>
```

Comment:
We should check `Next()` here as we can short circuit here if this is _already_ the end of the script.

Reply:
IIUC, we _only_ care about what this key value is (the only difference between the to local script and the to remote script). We have the pubkey in the sign desc, so we can check that (after applying the tweaks to it). 

This should shrink the function significantly.

Reply:
The difference is both what key and the timelock. If we want to use keys to discriminate we need to be able to tell the difference between the delay key and the non-delay key. The sign descriptor *is* present but it will supply us with the key in either case. We need a way to ask the question "is this the delay key or not". Unless I'm missing something, that context isn't currently available and so we would need to thread it in.

Reply:
Ir's possible for the remote party to propose that our CSV delay is 1 and we'll accept it, so the time lock works, but only if we add protection in lnd to reject 1 block CSVs. 

Good point about the delay vs the non-delay key! 

In this case, I think we can set more attributes here, namely the local channel config: https://github.com/lightningnetwork/lnd/blob/ea8f958dd513818d881de0f5e24627e4330e927a/contractcourt/commit_sweep_resolver.go#L509-L519. 

If we look at the two cases of local vs remote force close, we use a disticnt payment base point: 
  * Local force close uses the delay base point: https://github.com/lightningnetwork/lnd/blob/71ba355d903f01990718866c358c2ccac9402891/lnwallet/channel.go#L7690
  * Remote force close uses the payment base point: https://github.com/lightningnetwork/lnd/blob/71ba355d903f01990718866c358c2ccac9402891/lnwallet/channel.go#L6723

So I think we can get away with _only_ checking the `KeyDesc` contained in the `SignDesciptor` against either the delay or payment base point. This doesn't make any assumptions about the CSV delay that we'll accept. Instead, it relies on the fact that we'll only make a resolver once the commitment transaction has been confirmed on chain.

Reply:
In this case do we compute the key by using the basepoint + the per commitment point? I found this in BOLT 3:

```pubkey = basepoint + SHA256(per_commitment_point || basepoint) * G```

Reply:
Ok it looks like `SignDescriptor` has a field called `SingleTweak` that may be what I'm looking for here.

Reply:
Nvm. I figured out how to do it in the way you described.